Wrappers/Python/test/test_io.py

```diff
@@ -362,4 +362,70 @@ def TIFF_compression_test(self, compression, channels=1):
         np.testing.assert_array_equal(tmp, read_array)
 
         
+class TestRAW(unittest.TestCase):
+    def setUp(self) -> None:
+        # self.logger = logging.getLogger('cil.io')
+        # self.logger.setLevel(logging.DEBUG)
+        self.cwd = os.path.join(os.getcwd(), 'rawtest')
+        os.mkdir(self.cwd)
+
+    def tearDown(self) -> None:
+        shutil.rmtree(self.cwd)
+
+    def test_raw_nocompression_0(self):
+        self.RAW_compression_test(None,1)
+    
+    def test_raw_compression_0(self):
+        self.RAW_compression_test('uint8',1)
+
+    def test_raw_compression_1(self):
+        self.RAW_compression_test('uint16',1)
+
+    def test_raw_nocompression_1(self):
+        self.RAW_compression_test(None,1)
+    
+    def test_raw_compression_3(self):
+        with self.assertRaises(ValueError) as context:
+            self.RAW_compression_test(12,1)
+
+    def RAW_compression_test(self, compression, channels=1):
+        X=4
+        Y=5
+        Z=6
+        C=channels
+        if C == 1:
+            ig = ImageGeometry(voxel_num_x=4, voxel_num_y=5, voxel_num_z=6)
+        else:
+            ig = ImageGeometry(voxel_num_x=4, voxel_num_y=5, voxel_num_z=6, channels=C)
+        data = ig.allocate(0)
+        data.fill(np.arange(X*Y*Z*C).reshape(ig.shape))
+
+        from cil.io import utilities
+        compress = utilities.get_compress(compression)
+        dtype = utilities.get_compressed_dtype(data.array, compression)
+        scale, offset = utilities.get_compression_scale_offset(data.array, compression)
+        if C > 1:
+            assert data.ndim == 4
+        fname = os.path.join(self.cwd, "unittest.raw")
+        from cil.io import RAWFileWriter
+        writer = RAWFileWriter(data=data, file_name=fname, compression=compression)
+        writer.write()
+        
+        read_array = np.fromfile(fname, dtype=dtype)
```

Comment:
My understanding is that here you are setting the dtype yourself. Shouldn't you be reading the config file that the RAW writer created to get the datatype? Or you could do this in a separate test.

Also you could add another unit test for the Raw writer, including getting the other info from the config file.

Reply:
I implemented the test to reload the data from the `ini` file. I also added info about it on the docstring